docs/introduction.md

```diff
@@ -2,150 +2,449 @@
+    You may have noticed that `action_toggle_dark` doesn't do anything to explicitly change the _screen_, and yet hitting ++d++ updates the terminal. Textual is able to detect changes that should update the screen.
 
-The next example demonstrates handling events. Try running `intro02.py` in the `docs/examples/introduction`:
 
-```python title="intro02.py"
---8<-- "docs/examples/introduction/intro02.py"
+```python title="stopwatch01.py" hl_lines="22-24"
+--8<-- "docs/examples/introduction/stopwatch01.py"
 ```
 
-When you run this app you should see a blue screen in your terminal, like the following:
+The last few lines create an instance of the app at the module scope. Followed by a call to `run()` within a `__name__ == "__main__"` block. This is so that we could import `app` if we want to. Or we could run it with `python stopwatch01.py`. 
+
+## Designing a UI with widgets
+
+The header and footer are builtin widgets. For our Stopwatch application we will need to build custom widgets.
+
+Let's sketch out a design for our app:
 
-```{.textual path="docs/examples/introduction/intro02.py"}
+<div class="excalidraw">
+--8<-- "docs/images/stopwatch.excalidraw.svg"
+</div>
 
+We will need to build a `Stopwatch` widget composed of the following _child_ widgets:
+
+- A "Start" button
+- A "Stop" button
+- A "Reset" button
+- A time display
+
+Textual has a builtin `Button` widget which takes care of the first three components. All we need to build is the time display widget which will show the elapsed time in HOURS:MINUTES:SECONDS format, and the stopwatch widget itself.
+
+Let's add those to the app. Just a skeleton for now, we will add the rest of the features as we go.
+
+```python title="stopwatch02.py" hl_lines="3 6-7 10-18 28"
+--8<-- "docs/examples/introduction/stopwatch02.py"
 ```
 
-If you hit any of the number keys ++0++-++9++, the background will change color and you should hear a beep. As before, pressing ++ctrl+c++ will exit the app and return you to your prompt.
+### Extending widget classes
 
-!!! note
+We've imported two new widgets in this code: `Button`, which creates a clickable button, and `Static` which is a base class for a simple control. We've also imported `Container` from `textual.layout`. As the name suggests, `Container` is a Widget which contains other widgets. We will use this container to create a scrolling list of stopwatches.
+
+We're extending Static as a foundation for our `TimeDisplay` widget. There are no methods on this class yet. 
+
+The Stopwatch class also extends Static to define a new widget. This class has a `compose()` method which yields its child widgets, consisting of three `Button` objects and a single `TimeDisplay`. These are all we need to build a stopwatch as in the sketch.
+
+The Button constructor takes a label to be displayed in the button ("Start", "Stop", or "Reset"). Additionally some of the buttons set the following parameters:
```

Comment:
widget vs Widget vs `Widget`

Are there really three levels? “widget” should be the English noun and `Widget` is the Python class. “Widget” (capitalised, no backticks) can be dropped, right?
Then, it is “just” a matter of being careful and consistent with all other words.
Or not! If not, let me know and I will stop pointing these things out.

Reply:
If its referring to the class, then it probably should be backticked.